### DISCUSSION TITLE
Use the framework with firewlal configuration file

### DISCUSSION THREAD

**Discussion by BatD2** [PaloAltoNetworks/pan-os-python#426] [Q&A]
I have a script which is doing report on multple Panorama securirty policies. It is working ok using the API, but I need run it offline using the panorama config file. 
My understanding is that I should be able to do it using the refresh_from_xml() module, but I am keep on getting errors when using it. 

 As an example refreshall() is working fine:
`device_groups = panorama.DeviceGroup.refreshall(pano, add=True)`

but refresh_from_xml returns differenet errors: 

`device_groups = panorama.DeviceGroup.refreshall_from_xml(xml = 'running-config.xml', refresh_children = True)
TypeError: refreshall_from_xml() missing 1 required positional argument: 'self'`

Did you have any success running the framwork with a config file? Any guidance will be helpful.

**Comment by BatD2** [comment]
Can someone help?

**Comment by shinmog** [comment]
Will happen eventually, but this will require a bit of plumbing for pan-os-python to support.
